From 03f8af7daf70bf08dbcbc5c04ebe63fad616fb76 Mon Sep 17 00:00:00 2001
From: correctmost <11866-correctmost@users.noreply.gitlab.xfce.org>
Date: Sat, 20 Sep 2025 17:31:01 -0400
Subject: [PATCH] Fix criticals in g_file_info_get_name with trashed files

Per the enumerate_children docs, G_FILE_ATTRIBUTE_STANDARD_NAME
attributes must be gathered to avoid criticals like this when
iterating over the files:

GFileInfo created without standard::name
file ../gio/gfileinfo.c: line 1713 (g_file_info_get_name): should not be reached
---
 src/xfdesktop-special-file-icon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/xfdesktop-special-file-icon.c b/src/xfdesktop-special-file-icon.c
index 2c10807f2..0a4262634 100644
--- a/src/xfdesktop-special-file-icon.c
+++ b/src/xfdesktop-special-file-icon.c
@@ -455,7 +455,7 @@ xfdesktop_special_file_icon_update_trash_count(XfdesktopSpecialFileIcon *special
      * drive that isn't mounted.
      */
     enumerator = g_file_enumerate_children(special_file_icon->file,
-                                           G_FILE_ATTRIBUTE_ACCESS_CAN_DELETE,
+                                           G_FILE_ATTRIBUTE_ACCESS_CAN_DELETE "," G_FILE_ATTRIBUTE_STANDARD_NAME,
                                            G_FILE_QUERY_INFO_NONE,
                                            NULL,
                                            NULL);
-- 
GitLab


